src/librustc/hir/intravisit.rs

```diff
@@ -607,9 +607,11 @@ pub fn walk_ty<'v, V: Visitor<'v>>(visitor: &mut V, typ: &'v Ty) {
             }
             visitor.visit_lifetime(lifetime);
         }
-        TyImplTraitExistential(item_id, def_id, ref lifetimes) => {
+        TyImplTraitExistential(_, def_id, ref lifetimes) => {
+            // we are not recursing into the `existential` item, because it is already being visited
+            // as part of the surrounding module. The `NodeId` just exists so we don't have to look
+            // it up everywhere else in the compiler
             visitor.visit_def_mention(Def::Existential(def_id));
-            visitor.visit_nested_item(item_id);
```

Comment:
Hang on, if the `NodeId` is pointing to the same thing as the `DefId`, why do we keep both?!
We only really need the `DefId` for most of the compilation, AFAIK, and the occasional need to get the `NodeId` from the `DefId` shouldn't be enough to keep the `NodeId` in the HIR.

Reply:
I have further refactorings planned, this entire variant is getting removed ASAP, so I didn't feel very motivated to make it idiomatic

Reply:
Oh, huh, are you going with a fake path?

Reply:
Yea, not very fake though. The items are already real and in the parent of the function.

Reply:
Yeah, I mean just not part of the original AST. Seems good!